app/models/invoice.rb

```diff
@@ -106,20 +106,66 @@ def subscription_fees(subscription_id)
   end
 
   def recurring_fees(subscription_id)
-    subscription_fees(subscription_id).joins(charge: :billable_metric)
-      .merge(BillableMetric.recurring_count_agg)
+    subscription_fees(subscription_id)
+      .joins(charge: :billable_metric)
+      .where(billable_metric: { recurring: true })
+      .where(billable_metric: { aggregation_type: [1, 3] })
+      .where(charge: { pay_in_advance: false })
   end
 
   def recurring_breakdown(fee)
-    result = BillableMetrics::Aggregations::RecurringCountService.new(
+    service = case fee.charge.billable_metric.aggregation_type.to_sym
+              when :sum_agg
+                 BillableMetrics::Breakdown::SumService
+              when :unique_count_agg
+                 BillableMetrics::Breakdown::UniqueCountService
+              else
+                raise(NotImplementedError)
+              end
```

Comment:
Layout/EndAlignment: `end` at 124, 14 is not aligned with `service = case` at 117, 4.